src/subdomains/core/buy-crypto/routes/buy/buy.controller.ts

```diff
@@ -183,6 +188,45 @@ export class BuyController {
     };
   }
 
+  @Put('/transaction/:id/invoice')
+  @ApiBearerAuth()
+  @UseGuards(AuthGuard(), new RoleGuard(UserRole.USER), IpGuard, UserActiveGuard)
+  @ApiOkResponse({ type: InvoiceDto })
+  async generateInvoiceFromTransaction(@GetJwt() jwt: JwtPayload, @Param('id') id: string): Promise<InvoiceDto> {
+    const transaction = await this.transactionService.getTransactionById(+id, {
+      user: { userData: true },
+      buyCrypto: { outputAsset: true, buy: true },
+    });
+
+    if (!transaction) throw new BadRequestException('Transaction not found');
```

Comment:
I think I would add a check if the transaction isComplete. I would say we only return Invoices if amlCheck is `Pass` and the transaction is completed, or? 🤔  (same for sell, swap and rewards)